mne/decoding/csp.py

```diff
@@ -652,3 +656,150 @@ def _ajd_pham(X, eps=1e-6, max_iter=15):
+        in CSP space. Defaults to 'average_power'.
+
+    Attributes
+    ----------
+    ``filters_`` : ndarray, shape (n_components, n_channels)
+        If fit, the SPoC spatial filters, else None.
+    ``patterns_`` : ndarray, shape (n_components, n_channels)
+        If fit, the SPoC spatial patterns, else None.
+    ``mean_`` : ndarray, shape (n_components,)
+        If fit, the mean squared power for each component.
+    ``std_`` : ndarray, shape (n_components,)
+        If fit, the std squared power for each component.
+
+    See Also
+    --------
+    mne.preprocessing.Xdawn, CSP
+
+    References
+    ----------
+    [1] Dahne, S., Meinecke, F. C., Haufe, S., Hohne, J., Tangermann, M.,
+        Muller, K. R., & Nikulin, V. V. (2014). SPoC: a novel framework for
+        relating the amplitude of neuronal oscillations to behaviorally
+        relevant parameters. NeuroImage, 86, 111-122.
+    """
+
+    def __init__(self, n_components=4, reg=None, log=None,
+                 transform_into='average_power'):
+        """Init of SPoC."""
+        super(SPoC, self).__init__(n_components=n_components, reg=reg, log=log,
+                                   cov_est="epoch",
+                                   transform_into=transform_into)
+        delattr(self, 'cov_est')
+
+    def fit(self, X, y):
+        """Estimate the SPoC decomposition on epochs.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_epochs, n_channels, n_times)
+            The data on which to estimate the SPoC.
+        y : array, shape (n_epochs,)
+            The class for each epoch.
+
+        Returns
+        -------
+        self : instance of SPoC
+            Returns the modified instance.
+        """
+        if not isinstance(X, np.ndarray):
+            raise ValueError("X should be of type ndarray (got %s)."
+                             % type(X))
+        self._check_Xy(X, y)
+
+        if len(np.unique(y)) < 2:
+            raise ValueError("y must have at least two distinct values.")
+
+        # The following code is direclty copied from pyRiemann
+
+        # Normalize target variable
+        target = y.astype(np.float64)
+        target -= target.mean()
+        target /= target.std()
+
+        n_epochs, n_channels = X.shape[:2]
+
+        # Estimate single trial covariance
+        covs = np.empty((n_epochs, n_channels, n_channels))
+        for ii, epoch in enumerate(X):
+            covs[ii] = _regularized_covariance(epoch, reg=self.reg)
+
+        C = covs.mean(0)
+        Cz = np.mean(covs * target[:, np.newaxis, np.newaxis], axis=0)
+
+        # solve eigenvalue decomposition
+        evals, evecs = linalg.eigh(Cz, C)
+        evals = evals.real
+        evecs = evecs.real
+        # sort vectors
+        ix = np.argsort(np.abs(evals))[::-1]
+
+        # sort eigenvectors
+        evecs = evecs[:, ix].T
+
+        # spatial patterns
+        self.patterns_ = linalg.pinv(evecs).T  # n_components x n_channels
```

Comment:
... or if this is to take care of some rank deficiency, you could/should operate on `evals` directly to determine the cutoff